packages/opentelemetry-core/src/platform/browser/ShutdownNotifier.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+export function afterGlobalShutdown(cb: () => void = () => {}): void {
+  window.removeEventListener('unload', cb);
+}
+
+export function onGlobalShutdown(cb: () => void): void {
+  window.addEventListener('unload', () => {
+    afterGlobalShutdown(cb);
+  });
+}
+
+export function globalShutdownTestHelper(cb: () => void): void {
```

Comment:
Seems dangerous to export a function which does `window.close()`. Is there no way to put this functionality directly in the test files?

Reply:
The reason I put the function in this file was so that the proper functionality (window.close vs process.kill) would be exported for each test case. I can use try/catch in place of this and put the function directly in the test case if that makes more sense?

Reply:
I am just hesitant to export anything, even internally, which crashes the whole process. Seems dangerous.

Reply:
@obecny do you have an opinion on that?

Reply:
you could add that for you package only and don't export outside but use it internally only.

Reply:
It's probably fine to keep it here. Can you prefix with a _ so it is clear that it is "private" even though it is exported?